File: Python/inputs.py
import os
import settings


def build_filepath(year, day, use_examples):
    root_path = str(settings.input_path) + str(year) + str(settings.slash)
    example_dir = str(settings.example_dir) if use_examples else ""
    filename = "Day" + two_digit_day(day) + ".txt"
    filepath = root_path + example_dir + filename
    try:
        os.path.exists(filepath)
    except FileNotFoundError as e:
        raise FileNotFoundError(e)
    else:
        return filepath


def two_digit_day(day):
    # Enforce leading zero on days 1-9 for package and file names
    return "0" + str(day) if len(str(day)) == 1 else str(day)


def list_strings(filepath, separator):
    with open(filepath) as input:
        return list(map(str, input.read().split(separator)))


def list_ints(filepath, separator):
    with open(filepath) as input:
        return list(map(int, input.read().split(separator)))

File: CSharp/App/Inputs.cs
namespace AdventOfCode;
public static class Inputs{
    public static string BuildFilepath(int year, int day, bool useExamples){
        //Get the relevant filepath for the target OS from the configuration
        string inputPath = Settings.InputPath;
        //Expand out any environment variables in the path (e.g. %UserProfile% and %HOME%)
        string expandedPath = Environment.ExpandEnvironmentVariables(inputPath);
        // Determine the correct slash for filepaths in the target OS
        string os = Settings.Os;
        string slash = (os == "Windows") ? "\\" : "/";
        // If using example inputs, add the Examples subdirectory to the path
        string exampleDirectory = (useExamples == true) ? $"Examples{slash}" : "";
        // Format the filename to include a leading zero if the current day is less than 10
        string fileName = day < 10 ? $"Day0{day}.txt" : $"Day{day}.txt";
        // Construct the final filepath from the components above
        return $"{expandedPath}{slash}{year}{slash}{exampleDirectory}{fileName}";
    }
    public static List<string> ListLines(string filepath){
        return File.ReadAllLines(filepath).ToList();
    }
    public static List<string> ListStrings(string filepath, string separator){
        return File.ReadAllText(filepath).Split(separator).ToList();
    }
    public static List<int> ListInts(string filepath, string separator){
        return File.ReadAllText(filepath).Split(separator).Select(int.Parse).ToList();
    }
}